sdk/communication/azure-communication-jobrouter/azure/communication/jobrouter/aio/_router_administration_client_async.py

```diff
@@ -0,0 +1,1427 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from urllib.parse import urlparse
+# pylint: disable=unused-import,ungrouped-imports
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+    Union,
+    overload
+)
+
+from azure.core.tracing.decorator import distributed_trace
+from azure.core.tracing.decorator_async import distributed_trace_async
+from azure.core.async_paging import AsyncItemPaged
+
+from .._shared.user_credential_async import CommunicationTokenCredential
+from .._shared.utils import parse_connection_str, get_authentication_policy
+
+from .._generated._serialization import Serializer  # pylint:disable=protected-access
+from .._generated.aio import AzureCommunicationJobRouterService
+from .._generated.models import (
+    ClassificationPolicy,
+    DistributionPolicy,
+    ExceptionPolicy,
+    ClassificationPolicyItem,
+    DistributionPolicyItem,
+    ExceptionPolicyItem,
+    ExceptionRule,
+    BestWorkerMode,
+    LongestIdleMode,
+    RoundRobinMode,
+    StaticQueueSelectorAttachment,
+    ConditionalQueueSelectorAttachment,
+    RuleEngineQueueSelectorAttachment,
+    PassThroughQueueSelectorAttachment,
+    WeightedAllocationQueueSelectorAttachment,
+    StaticWorkerSelectorAttachment,
+    ConditionalWorkerSelectorAttachment,
+    RuleEngineWorkerSelectorAttachment,
+    PassThroughWorkerSelectorAttachment,
+    WeightedAllocationWorkerSelectorAttachment,
+    StaticRule,
+    ExpressionRule,
+    FunctionRule,
+)
+from .._models import (
+    JobQueue,
+    JobQueueItem,
+)
+
+from .._version import SDK_MONIKER
+
+_SERIALIZER = Serializer()
+_SERIALIZER.client_side_validation = False
+
+
+class RouterAdministrationClient(object):  # pylint: disable=client-accepts-api-version-keyword,too-many-public-methods,too-many-lines
+    """A client to interact with the AzureCommunicationService JobRouter service.
+
+    This client provides operations to create, update, list and delete the following entities: classification policy,
+    exception policy, distribution policy and queue.
+
+    :param str endpoint:
+        The endpoint of the Azure Communication resource.
+    :param CommunicationTokenCredential credential:
+        The credentials with which to authenticate
+    """
+
+    def __init__(
+            self,
+            endpoint: str,
+            credential: CommunicationTokenCredential,
```

Comment:
Do we have an async version of this credential?

Reply:
I have changed this to use `TokenCredential` and `AsyncTokenCredential` from `azure.core.credentials` and `_async` respectively

Reply:
We currently only support `AzureKeyCredential`